src/ssl/bio.cc

```diff
@@ -714,6 +714,9 @@ applyTlsDetailsToSSL(SSL *ssl, Security::TlsDetails::Pointer const &details, Ssl
                 if (!strCiphers.isEmpty())
                     strCiphers.append(":");
                 strCiphers.append(SSL_CIPHER_get_name(c));
+            } else {
+                debugs(83, 7, "ignoring " << (Security::Greased(cipherId) ? "GREASEd" : "unsupported") <<
+                       " cipher: " << cipherId);
```

Comment:
@chtsanti, do we need to keep adding unsupported and GREASEd cipher IDs to `details->ciphers`? This code does not seem to need them. I know adjustSSL() checks ciphers, but there we could just remember whether we saw any unsupported ones during parsing -- we already do the same for extensions with `details->unsupportedExtensions`.

Reply:
What if in the future implement ciphers based acls or an access list which allow/deny special ciphers? In these cases maybe it is required to tke in account GREASEd values.

Reply:
Thank you for a prompt answer. I agree that such a future is possible, perhaps even likely. I will be ableto move forward with that information in mind.

FWIW, I do not like the inconsistency between our approaches to dealing with unsupported and GREASEd cipher IDs and dealing with unsupported and GREASEd TLS versions. However, resolving that inconsistency is probably outside this issue scope. I do not plan on doing that here.